framework/PostProcessors/RavenOutput.py

```diff
@@ -37,218 +37,223 @@ class RavenOutput(PostProcessor):
+#          desiredNode = self._readPath(pivotStep,path,fileName)
+#          realization['outputs'][name] = float(desiredNode.text)
+#        outputDict['realizations'].append(realization)
+#    else:
+#      # each ID results in a realization for the requested attributes
+#      for id,fileDict in self.files.items():
+#        realization = {'inputs':{'ID':id},'outputs':{},'metadata':{'loadedFromRavenFile':str(fileDict['fileObject'])}}
+#        for varName,path in fileDict['paths'].items():
+#          #read the value from the file's XML
+#          root,_ = xmlUtils.loadToTree(fileDict['fileObject'].getAbsFile())
+#          desiredNode = self._readPath(root,path,fileDict['fileObject'].getAbsFile())
+#          realization['outputs'][varName] = float(desiredNode.text)
+#        outputDict['realizations'].append(realization)
+#    return outputDict
+#
+#  def collectOutput(self, finishedJob, output):
+#    """
+#      Function to place all of the computed data into the output object
+#      @ In, finishedJob, JobHandler External or Internal instance, A JobHandler object that is in charge of running this post-processor
+#      @ In, output, dataObjects, The object where we want to place our computed results
+#      @ Out, None
+#    """
+#    evaluation = finishedJob.getEvaluation()
+#    if isinstance(evaluation, Runners.Error):
+#      self.raiseAnError(RuntimeError, "No available output to collect (run possibly not finished yet)")
+#
+#    outputDictionary = evaluation[1]
+#    realizations = outputDictionary['realizations']
+#    for real in realizations:
+#      for key in output.getParaKeys('inputs'):
+#        if key not in real['inputs'].keys():
+#          self.raiseAnError(RuntimeError, 'Requested input variable '+key+' has not been extracted. Check the consistency of your input')
+#        output.updateInputValue(key,real['inputs'][key])
+#      for key in output.getParaKeys('outputs'):
+#        if key not in real['outputs'].keys():
+#          self.raiseAnError(RuntimeError, 'Requested output variable '+key+' has not been extracted. Check the consistency of your input')
+#        output.updateOutputValue(key,real['outputs'][key])
+#      for key,val in real['metadata'].items():
+#        output.updateMetadata(key,val)
+#
+#  def _readPath(self,root,inpPath,fileName):
+#    """
+#      Reads in values from XML tree.
+#      @ In, root, xml.etree.ElementTree.Element, node to start from
+#      @ In, inPath, string, |-separated list defining path from root (not including root)
+#      @ In, fileName, string, used in error
+#      @ Out, desiredNode, xml.etree.ElementTree.Element, desired node
+#    """
+#    #improve path format
+#    path = '/'.join(c.strip() for c in inpPath.strip().split('/'))
+#    desiredNode = xmlUtils.findPath(root,path)
+#    if desiredNode is None:
+#      self.raiseAnError(RuntimeError,'Did not find "%s/%s" in file "%s"' %(root.tag,path,fileName))
+#    return desiredNode
```

Comment:
Do we need all this commented lines?

Reply:
Yes, otherwise it counts against us in the coverage check; however, the code is likely to be very useful, as it still applies to the XML form of the general metadata.